mne/export/_edf.py

```diff
@@ -309,24 +324,15 @@ def _export_raw(fname, raw, physical_range, add_ch_type):
                 raw.annotations.duration,
                 raw.annotations.ch_names,
             ):
-                # annotations are written in terms of 100 microseconds
-                onset = onset * 10000
-                duration = duration * 10000
+                # annotations are written in terms of seconds
                 if ch_names:
                     for ch_name in ch_names:
-                        if (
+                        try:
                             hdl.writeAnnotation(onset, duration, desc + f"@@{ch_name}")
-                            != 0
-                        ):
+                        except Exception:
                             raise RuntimeError(
                                 f"writeAnnotation() returned an error "
-                                f"trying to write {desc}@@{ch_name} at {onset} "
+                                f"trying to write {desc} at {onset} "
                                 f"for {duration} seconds."
                             )
-                else:
-                    if hdl.writeAnnotation(onset, duration, desc) != 0:
-                        raise RuntimeError(
-                            f"writeAnnotation() returned an error "
-                            f"trying to write {desc} at {onset} "
-                            f"for {duration} seconds."
-                        )
+    del hdl
```

Comment:
I'm not sure if it is. I just included it because in the [documentation ](https://pyedflib.readthedocs.io/en/latest/_modules/pyedflib/highlevel.html#write_edf)for the pyedflib high level functions they include it:

```
    with pyedflib.EdfWriter(edf_file, n_channels=n_channels, file_type=file_type) as f:
        f.setDatarecordDuration(int(100000 * block_size))
        f.setSignalHeaders(signal_headers)
        f.setHeader(header)
        f.writeSamples(signals, digital=digital)
        for annotation in annotations:
            f.writeAnnotation(*annotation)
    del f
```

Reply:
@wmvanvliet Thank you for the suggestions! The anotherthing I really wanted help/opinions on is getting the tolerances to be lower and getting the [setDatarecordDuration](https://pyedflib.readthedocs.io/en/latest/ref/edfwriter.html#pyedflib.EdfWriter.setDatarecordDuration) function to match the original implementation so that the sampling rate matches the original implementation

Reply:
regarding `del f`, I really don't think it's necessary since we're at the end of the function already: when the function returns, `f` should be cleaned up automatically.

Reply:
> Could you explain a bit more about the tolerances and need to set setDatarecordDuration? Sorry, I'm note completely up to date about the details of the EDF/BDF file format.

Sure @wmvanvliet, basically when I tested my first code, the tolerances were way too high:
<img width="383" alt="image" src="https://github.com/mne-tools/mne-python/assets/20428109/ab54eafd-49a9-4815-9dc7-7b5e1bdc8bbb">
The current implementation expects tolerances of `1e-5`.  Also, the [setDatarecordDuration](https://pyedflib.readthedocs.io/en/latest/ref/edfwriter.html#pyedflib.EdfWriter.setDatarecordDuration) function basically lets you have a non-int sampling rate. There are some problems with the documentation ([see this issue)](https://github.com/holgern/pyedflib/issues/242), but I think I figured that out. The function is supposed to give the same sampling rate as the original file but instead, it is slightly lower. 
Original:
<img width="356" alt="image" src="https://github.com/mne-tools/mne-python/assets/20428109/205baac7-bc5f-4dc8-a72d-da0ac36c6b1f">
EDF export:
<img width="229" alt="image" src="https://github.com/mne-tools/mne-python/assets/20428109/6f040e6f-b850-4c78-9937-ac7a0d71ad6d">

I suspect this is one reason why the tolerances are different, but it can't entirely explain the problem as the tolerances are too high even with an integer sampling rate.

Reply:
The current precision is 1.110142833033462e-08 V, or 0.011101428330334619 µV – so we should be good?

Reply:
From my testing I thought it was `1e-5` µV, but I could have been wrong. Also, the test expects a tolerance of `1e-5` for the timepoints as well:
```
    # Due to the data record duration limitations of EDF files, one
    # cannot store arbitrary float sampling rate exactly. Usually this
    # results in two sampling rates that are off by very low number of
    # decimal points. This for practical purposes does not matter
    # but will result in an error when say the number of time points
    # is very very large.
    assert_allclose(raw.times, raw_read.times[:orig_raw_len], rtol=0, atol=1e-5)
```
This is why the sampling rate is a problem because it changes the number of time points and therefore the value. Maybe we just need to change the tests?